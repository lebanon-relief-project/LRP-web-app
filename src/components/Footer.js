import React from "react";
import styled from "styled-components";
import FooterLink from "./FooterLink";
import FooterImageLink from "./FooterImageLink";
import { PageContainer } from "../styles/GlobalStyles";
import colours from "../styles/Colours";

const Footer = () => {
  return (
    <StyledSection>
      <PageContainer>
        <CardWrapper>
          <FooterLink
            title="Iâ€™m looking for help"
            links={["Terms & Conditions", "Privacy policy"]}
          />
          <FooterLink title="About us" links={["What we do", "Contact us"]} />
          <FooterLink
            title="Partnerships"
            links={["How to partner with us", "Working with us"]}
          />
          <FooterImageLink text="Lorem lobortis aliquam leo nisi vita" />
        </CardWrapper>
      </PageContainer>
    </StyledSection>
  );
};

const CardWrapper = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: space-evenly;
`;

const StyledSection = styled.section`
  background: ${colours.grey};
  margin: 0;

  min-height: 5%;
`;

export default Footer;
