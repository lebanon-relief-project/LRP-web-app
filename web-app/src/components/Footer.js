import React from "react";
import styled from "styled-components";
import FooterLink from "./FooterLink";
import colours from "../styles/Colours";
import devices from "../styles/Devices";

const Footer = () => {
  return (
    <StyledSection>
      <CardWrapper>
        <FooterLink
          title="Lebanon Relief Network"
          links={["Terms & Conditions", "Privacy policy"]}
        />
        <FooterLink title="About us" links={["What we do", "Contact us"]} />
        <FooterLink
          title="Partnerships"
          links={["How to partner with us", "Working with us"]}
        />
        <CircularLinkWrapper>
          <CircularLink />
          <CircularLink />
          <CircularLink />
        </CircularLinkWrapper>
      </CardWrapper>
    </StyledSection>
  );
};

const CircularLink = styled.span`
  height: 2.5rem;
  width: 2.5rem;
  background-color: #bbb;
  border-radius: 50%;
  display: inline-block;
  margin: 0.5rem;
`;

const CircularLinkWrapper = styled.div`
  display: flex;
  padding: 1rem;
  margin: 2rem 0;
`;

const CardWrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-evenly;

  @media (max-width: ${devices.mobile}) {
    flex-direction: column;
    align-items: center;
  }
`;

const StyledSection = styled.section`
  background: ${colours.grey};
  margin: 0;
  min-height: 5%;
`;

export default Footer;
