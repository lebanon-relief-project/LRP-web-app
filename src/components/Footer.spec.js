import React from "react";
import { render } from "@testing-library/react";
import Footer from "./Footer";
import "jest-styled-components";

describe("the Footer component", () => {
  let container;
  it("should match the snapshot", () => {
    ({ container } = renderFooter());

    expect(container).toMatchSnapshot();
  });
});

const renderFooter = () => {
  return render(<Footer />);
};
