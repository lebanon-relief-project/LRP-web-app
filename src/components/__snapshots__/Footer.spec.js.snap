// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Footer component should match the snapshot 1`] = `
<div>
  <section
    class="sc-iCoHVE jAMKrk"
  >
    <div
      class="sc-gKAblj jWlcXc"
    >
      <div
        class="sc-gtssRu hNSdtc"
      >
        <h2
          class="sc-bdnylx gpJKAy"
        >
          Lebanon Relief Network
        </h2>
        <ul
          class="sc-dlnjPT cQMtRp"
        >
          <li
            class="sc-hKFyIo igLYdx"
          >
            Terms & Conditions
          </li>
          <li
            class="sc-hKFyIo igLYdx"
          >
            Privacy policy
          </li>
        </ul>
      </div>
      <div
        class="sc-gtssRu hNSdtc"
      >
        <h2
          class="sc-bdnylx gpJKAy"
        >
          About us
        </h2>
        <ul
          class="sc-dlnjPT cQMtRp"
        >
          <li
            class="sc-hKFyIo igLYdx"
          >
            What we do
          </li>
          <li
            class="sc-hKFyIo igLYdx"
          >
            Contact us
          </li>
        </ul>
      </div>
      <div
        class="sc-gtssRu hNSdtc"
      >
        <h2
          class="sc-bdnylx gpJKAy"
        >
          Partnerships
        </h2>
        <ul
          class="sc-dlnjPT cQMtRp"
        >
          <li
            class="sc-hKFyIo igLYdx"
          >
            How to partner with us
          </li>
          <li
            class="sc-hKFyIo igLYdx"
          >
            Working with us
          </li>
        </ul>
      </div>
      <div
        class="sc-jSFkmK dzlDOT"
      >
        <span
          class="sc-eCApGN byfgNG"
        />
        <span
          class="sc-eCApGN byfgNG"
        />
        <span
          class="sc-eCApGN byfgNG"
        />
      </div>
    </div>
  </section>
</div>
`;
